### PR TITLE
Convert to .NET Standard project

### PR DESCRIPTION
**Pull request by tirth**
Hi Rex, it's Tirth from PMH. I mentioned to you at AAPM about updating to the new VS project system - here's the initial port for you to review. There are still a few questions to address, as well as making sure the tests run.

Specifically:

- which version of .NET Standard to target? 2.0 means no external dependencies except Microsoft.CSharp (for the dynamic usages, which could be refactored out)
- how much to bump the EvilDICOM version, should it be 3.0?

### CODE REVIEW

**Comment by feliwir** [comment]
Can this be merged please?

**Comment by tirth** [comment]
Just updated the project system a little more by refactoring the code generator, targeting netstandard2.0, and moving it into the main solution, and updated the generated classes using the most recent standard.

**Comment by feliwir** [comment]
@tirth i am not sure if the owner of this project is still active. Do you have a fork or something?

**Comment by tirth** [comment]
@feliwir unfortunately I don't have the time to maintain anything official on NuGet, but you can build from my GitHub fork

**Comment by dalingrin** [comment]
I'd suggest adding net45 to the <TargetFrameworks> in the csproj. Many of the computers our software runs on only has .Net Framework 4.5.x. By targetting both net45 and netstandard20, the library will remain compatible with all systems it was prior to this pull request while still getting all the cross-platform benefits of .net standard 2.0.

**Comment by feliwir** [comment]
@dalingri I don’t think that’s required. .NETStandard can be used by recent .NETFrameworks. NET Framework 4.5. E.g. it works on both Windows 7 & Windows 10 without any issues.

**Comment by tirth** [comment]
@feliwir [It looks like .NET Framework 4.5.* only supports up to .NET Standard 1.2](https://docs.microsoft.com/en-us/dotnet/standard/net-standard#net-implementation-support). 

@dalingrin I'll look into multitargeting it with netstandard1.2 or net45, I've (fortunately) been stuck with .NET Framework 4.6.1 so netstandard2.0 worked.

@rexcardan Will you have a chance to look over the PR any time soon? I've been adding some more functionality in our fork like a DICOMWeb client and IOD module writing helpers, and experimenting with Span-based reading for more throughput. It builds on the updated project layout here though so I can't quite submit the changes separately.

File: EvilDICOM/EvilDICOM/EvilDICOM.csproj [hunk 1]
```diff
@@ -1,301 +1,34 @@
...
-    <OutputPath>bin\Release\</OutputPath>
-    <DefineConstants>TRACE;NET45</DefineConstants>
-    <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
-    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
-  </PropertyGroup>
+<Project Sdk="Microsoft.NET.Sdk">
+
   <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (dalingrin) [inline EvilDICOM/EvilDICOM/EvilDICOM.csproj]:
```suggestion
    <TargetFrameworks>netstandard2.0;net45</TargetFrameworks>
```

File: EvilDICOM/EvilDICOM/EvilDICOM.csproj [hunk 2]
```diff
@@ -1,301 +1,34 @@
...
-    <Content Include="evilEye.ico" />
-  </ItemGroup>
-  <ItemGroup>
-    <Folder Include="Network\Services\CFind\" />
-  </ItemGroup>
+
   <ItemGroup>
-    <None Include="packages.config" />
-    <None Include="redion_sn.pfx" />
+    <PackageReference Include="Microsoft.CSharp" Version="4.5.0" />
```
Comment (dalingrin) [inline EvilDICOM/EvilDICOM/EvilDICOM.csproj]:
```suggestion
    <PackageReference Include="Microsoft.CSharp" Version="4.5.0" />
    <PackageReference Include="System.ValueTuple" Version="4.5.0" />
```

**Comment by rexcardan** [comment]
I have created a .net core release in 2.0.5.1

**Pull request closed without merging**